### PR TITLE
Add gallery example showing how to adjust line segment ends (caps and joints)

### PR DESCRIPTION
**Pull request by yvonnefroehlich**
**Description of proposed changes**

This PR aims to add a gallery example showing how to adjust line segment ends (caps and joints).
- [x] Add basic code
- [x] Combine to Subplot
- [x] Add docs
- [x] Fix typo in upstream GMT documentation https://github.com/GenericMappingTools/gmt/pull/8315

Orientated on the upstream GMT example: https://docs.generic-mapping-tools.org/dev/reference/features.html#tbl-penex

**Preveiw**: https://pygmt-dev--3015.org.readthedocs.build/en/3015/gallery/lines/line_segment_ends.html

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.
- [ ] Use underscores (not hyphens) in names of Python files and directories.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash command is:

- `/format`: automatically format and lint the code

### CODE REVIEW

File: examples/gallery/lines/line_segment_ends.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+"""
+Line segment caps and joints
+============================
+PyGMT offers different appearance of line segment caps and joints. The desired
+appearance can be set via the GMT default parameters :gmt-term:`PS_LINE_CAP`
+(``"butt"``, ``"round"``, or ``"square"`` [Default]) as well as :gmt-term:`PS_LINE_JOIN`
+(``"bevel"``, `"round"``, and ``"miter"`` [Default]) and :gmt-term:`PS_MITER_LIMIT`
```
Comment (yvonnefroehlich) [inline examples/gallery/lines/line_segment_ends.py:7]:
```suggestion
(``"bevel"``, ``"round"``, and ``"miter"`` [Default]) and :gmt-term:`PS_MITER_LIMIT`
```

File: examples/gallery/lines/line_segment_ends.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 250, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
+    # The change applies only to the code under the with statement
+    with pygmt.config(PS_LINE_CAP=line_cap):
+        # Plot a dashed line
+        fig.plot(x=x, y=y, pen="10p," + dict_col[line_cap] + ",20_20:0")
```
Comment (seisman) [inline examples/gallery/lines/line_segment_ends.py:41]: Make it more readable:
```suggestion
        # Plot a dashed line
        color = dict_col[line_cap]
        fig.plot(x=x, y=y, pen=f"10p,{color},20_20:0")
```

File: examples/gallery/lines/line_segment_ends.py [hunk 3]
```diff
@@ -0,0 +1,75 @@
+"""
+Line segment caps and joints
+============================
+PyGMT offers different appearance of line segment caps and joints. The desired
```
Comment (seisman) [inline examples/gallery/lines/line_segment_ends.py:4]:
```suggestion
PyGMT offers different appearances of line segment caps and joints. The desired
```

File: examples/gallery/lines/line_segment_ends.py [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+import pygmt
+
+# Set up dictionary for colors
+dict_col = {
+    "round": "green4",
+    "square": "steelblue",
+    "butt": "orange",
+    "miter": "steelblue",
+    "bevel": "orange",
+}
```
Comment (seisman) [inline examples/gallery/lines/line_segment_ends.py:22]: Are the different colors necessary? Maybe just use a single color for all dashed lines?

File: examples/gallery/lines/line_segment_ends.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    y = y - 20
+
+fig.shift_origin(yshift="-h")
+
+# -----------------------------------------------------------------------------
+# Bottom: PS_LINE_JOIN and PS_MITER_LIMIT
+
+# Create sample data
+x = np.array([[5, 95, 65], [80, 170, 140], [155, 235, 205]])
+y = np.array([[10, 70, 10], [10, 70, 10], [10, 70, 10]])
```
Comment (seisman) [inline examples/gallery/lines/line_segment_ends.py:57]: This sample data is a little too complicated. Maybe just define the data as
```
x = np.array([5, 95, 65])
y = np.array([10, 70, 10])
```
for other two lines, just use `x += 75`.

Comment (yvonnefroehlich) [reply examples/gallery/lines/line_segment_ends.py]: Yes, you are completely right! I totally overlooked this after changing to lines which are only shifted in x-direction in the second part! Done in https://github.com/GenericMappingTools/pygmt/pull/3015/commits/bdaf946b4b22323feecfe17c23f7975b5be9f41a.

File: examples/gallery/lines/line_segment_ends.py (hunk 4 above)
Comment (yvonnefroehlich) [reply examples/gallery/lines/line_segment_ends.py:22]: The idea regarding the colors was to have the same color for similar appearances of `PS_LINE_CAP` and `PS_LINE_JOIN`:
line_segment_ends_colors

File: examples/gallery/lines/line_segment_ends.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+# Create sample data
+x = np.array([30, 170])
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 260, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
+    # The change applies only to the code under the with statement
```
Comment (michaelgrund) [inline examples/gallery/lines/line_segment_ends.py:38]:
```suggestion
```

File: examples/gallery/lines/line_segment_ends.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+# Top: PS_LINE_CAP
+
+# Create sample data
+x = np.array([30, 170])
+y = np.array([70, 70])
+
+fig.basemap(region=[0, 260, 0, 100], projection="x1p", frame="rltb")
+
+for line_cap in ["butt", "round", "square"]:
+    # Change GMT default locally using a context manager
```
Comment (michaelgrund) [inline examples/gallery/lines/line_segment_ends.py:37]:
```suggestion
    # Change GMT default locally
```
Should be clear without an extended description what the with statement does.

File: examples/gallery/lines/line_segment_ends.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+# Create sample data
+x = np.array([5, 95, 65])
+y = np.array([10, 70, 10])
+
+fig.basemap(region=[0, 260, 0, 100], projection="x1p", frame="rltb")
+
+for line_join in ["bevel", "round", "miter"]:
+    # Change GMT default locally using a context manager
+    # The change applies only to the code under the with statement
```
Comment (michaelgrund) [inline examples/gallery/lines/line_segment_ends.py:63]:
```suggestion
```
I think we don't need this description twice within the example script.

**Review by seisman — APPROVED** [review]

**Pull request merged**